gcc/config/riscv/constraints.md

```diff
@@ -118,6 +118,162 @@
   (and (match_operand 0 "move_operand")
        (match_test "CONSTANT_P (op)")))
 
+;; CORE-V Constraints
+(define_constraint "PALU31"
+  "Checking for ALU clip if it's a power of 2"
+  (and (match_code "const_int")
+       (match_test "ival == 1073741823")))
```

Comment:
I think these can be replaced with a single constraint by using `pow2p_hwi`.

Reply:
Would something like this be what you expect?
```
int i;
for (i = 0; i <= 30; i++)
  if (pow2p_hwi (i) - 1 == ival)
    return true;
return false;
```